iphone/Classes/KrollContext.m

```diff
@@ -1617,29 +1628,67 @@ - (void)invalidateAllTimers
 
 - (NSUInteger)setIntervalFromArguments:(NSArray<JSValue *> *)arguments shouldRepeat:(BOOL)shouldRepeat
 {
-  NSMutableArray *callbackArgs = [arguments.mutableCopy autorelease];
+  NSMutableArray<JSValue *> *callbackArgs = [arguments.mutableCopy autorelease];
+  // Retain the callback
   JSValue *callbackFunction = [callbackArgs objectAtIndex:0];
   [callbackArgs removeObjectAtIndex:0];
-  double interval = [[callbackArgs objectAtIndex:0] toDouble] / 1000;
-  [callbackArgs removeObjectAtIndex:0];
+  JSVirtualMachine *vm = callbackFunction.context.virtualMachine;
```

Comment:
For SDK < 8, this should likely use the Ti prefixed JSCore calls, otherwise this could break apps not using JSCore so far.